src/state.rs

```diff
@@ -59,6 +71,29 @@ impl State {
             db: self.db.snapshot(),
             open_files: self.open_files.snapshot(),
             config: self.config.snapshot(),
+            beacon: Default::default(),
+        }
+    }
+}
+/// Struct which allows setting a callback - which can be triggered afterward
+/// by the function which has the reference.
+#[derive(Default)]
+pub struct Beacon {
```

Comment:
TBH you have simplified this so much that I would probably drop this type entirely and just stick that `Option<Box<dyn FnOnce() + Send>>`.